### PR TITLE
Changes to support Connext 6.1.0

### PR DESCRIPTION
**Pull request by marc-chiesa**
This PR includes the following changes to the API:

- added support for new Connext 6.1.0 features
- some method names changed to better reflect their purpose
- static constructors for specific mask type values have been converted to be ALL_CAPS static properties to reflect that they are like constants
- static constructors for specific object settings have been converted to static properties for consistency with this change that had previously been implemented in certain parts of the API (e.g. QosProvider.default)
- added submodules to support network capture and heap monitoring
- examples and tests have been updated to reflect the above changes

Other minor changes were made to the build system, including the following:

- support bdist_rpm option
- find_package for Python will now find same major version used to run setup.py

### CODE REVIEW

File: modules/connextdds/include/PyNamespaces.hpp [hunk 1]
```diff
@@ -1,45 +1,51 @@
 /*
  * (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
  *
- * RTI grants Licensee a license to use, modify, compile, and create derivative
+ * RTI grantsicensee aicense to use,odify, compile, and create derivative
```
Comment (hman523) [inline modules/connextdds/include/PyNamespaces.hpp:4]: Is this a typo here and the next few lines?

Comment (marc-chiesa) [reply modules/connextdds/include/PyNamespaces.hpp]: That was an oversight on a search/replace that inadvertently modified the header block. Fixed in 
0140bc1

File: modules/connextdds/src/rti/core/policy/DataReaderInstanceRemovalKind.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+namespace pyrti {
+
+template<>
+void process_inits<DataReaderInstanceRemovalKind>(py::module& m, ClassInitList& l)
+{
+    init_dds_safe_enum<
+            DataReaderInstanceRemovalKind_def>(m, "DataReaderInstanceRemovalKind ", [](py::object& o) {
+        py::enum_<DataReaderInstanceRemovalKind::type>(o, "DataReaderInstanceRemovalKind ")
+                .value("NO_INSTANCE",
+                       DataReaderInstanceRemovalKind ::type::NO_INSTANCE,
```
Comment (hman523) [inline modules/connextdds/src/rti/core/policy/DataReaderInstanceRemovalKind.cpp:30]:
```suggestion
                       DataReaderInstanceRemovalKind::type::NO_INSTANCE,
```

**Review by hman523 — APPROVED** [review]
I read through all of the files and found a few small errors that I have already commented on

**Pull request merged**